#### Part 1- cleaned data

In [1]:
# Dependencies and Setup
import matplotlib.pyplot as plt
import pandas as pd
import scipy.stats as st
import numpy as np
import seaborn as sns


#### Question 1:

#### How many total reservations are in this dataset (cleaned)? 
72,456

In [24]:
# Import the data/csv file & preview
df = pd.read_csv("cleaned_data2.csv")
df = df.dropna()
df = df.reset_index(drop=True)

original = pd.read_csv("cleaned_data2.csv")
original = df.dropna()
original = df.reset_index(drop=True)

df

Unnamed: 0  unit_id check_in_date check_out_date reservation_type  \
0               0    40630      1/1/2018       1/3/2018            guest   
1               1    40630      1/3/2018       1/7/2018            guest   
2               2    40630     1/11/2018      1/15/2018            guest   
3               3    40630     1/15/2018      1/18/2018            owner   
4               4    40630     1/20/2018      1/21/2018            guest   
...           ...      ...           ...            ...              ...   
72451       92536    21239      3/9/2019      3/13/2019            guest   
72452       92538    21239     3/21/2019      3/24/2019            guest   
72453       92540    21239      4/9/2019      4/12/2019            guest   
72454       92541    21239     4/12/2019      4/14/2019            guest   
72455       92542    21239     4/14/2019      4/16/2019            guest   

             reservation_source_type reservation_created_date  adult_guests  \
0                            unknown                 1/1/2018             2   
1                            unknown                 1/3/2018             2   
2                            unknown                9/29/2017             2   
3                            unknown                1/15/2018             2   
4                            unknown                1/17/2018             2   
...                              ...                      ...           ...   
72451  3rd Party Channel Reservation                2/17/2019             4   
72452  3rd Party Channel Reservation                3/20/2019             4   
72453  3rd Party Channel Reservation                 4/8/2019             2   
72454                         Vacasa                 4/8/2019             2   
72455  3rd Party Channel Reservation                 4/8/2019             6   

       child_guests  pets  total_paid  city_name   zip5 state_name_short  \
0                 0     0        1930       Vail  81657               CO   
1                 0     0        2969       Vail  81657               CO   
2                 0     0       12913       Vail  81657               CO   
3                 0     0          60       Vail  81657               CO   
4                 0     0         515       Vail  81657               CO   
...             ...   ...         ...        ...    ...              ...   
72451             0     0        1300  Galveston  77554               TX   
72452             0     0         661  Galveston  77554               TX   
72453             0     0         340  Galveston  77554               TX   
72454             1     0         245  Galveston  77554               TX   
72455             0     0         287  Galveston  77554               TX   

          region_name_short  
0       Vail / Beaver Creek  
1       Vail / Beaver Creek  
2       Vail / Beaver Creek  
3       Vail / Beaver Creek  
4       Vail / Beaver Creek  
...                     ...  
72451  The Texas Gulf Coast  
72452  The Texas Gulf Coast  
72453  The Texas Gulf Coast  
72454  The Texas Gulf Coast  
72455  The Texas Gulf Coast  

[72456 rows x 15 columns]

#### Question 2:

#### What's the number of units that had a reservation? 
2,125

In [3]:
df['unit_id'].value_counts()

9131     157
11033    146
6309     145
17653    143
40618    143
        ... 
31087      1
31237      1
9107       1
30653      1
17594      1
Name: unit_id, Length: 2125, dtype: int64

#### Question 3:

#### How many reservations in Vail?

Vail / Beaver Creek: 674 reservations

In [4]:
# Get count of reservations at Vail / Beaver Creek
# Reservations & Unit Countsd
region_unit = df.groupby(by="region_name_short")["unit_id"].nunique()
region_unit

region_name_short
Delaware                939
The Texas Gulf Coast    512
Vail / Beaver Creek     674
Name: unit_id, dtype: int64


#### Question 4: 

#### How many check-in's were there in Vail in March of 2019?

Need assistance completing this in Pandas. 

In [14]:
# Empty

#### Question 5: 

#### What % of reservations had children?

67.54% 

In [16]:
avg_child = df['child_guests'].mean()
avg_child

0.6753615987633874

#### Question 6:

#### What's the average number of children on a reservation? 

1

In [17]:
# Get value count of # of children guests 
child_sum = df['child_guests'].sum()
child_sum

48934

In [19]:
# Divide # of reservations / # of children 
res_by_child = np.true_divide(72456, 48935)
res_by_child

1.480658015735159

#### Question 7:

#### Complete table with Reservations & Unit Counts

   #### Reservations|  Units 
        
 Texas-     26,351  |  512
 
 Delaware-  20,373  |  939
 
 Vail-      25,732  |  674

In [20]:
# Reservations & Unit Counts
res_unit_counts = df.groupby(by="region_name_short")["unit_id"].nunique()
res_unit_counts

region_name_short
Delaware                939
The Texas Gulf Coast    512
Vail / Beaver Creek     674
Name: unit_id, dtype: int64

In [21]:
# Get count of reservations at Vail / Beaver Creek
vail_res = df['region_name_short'].value_counts()
vail_res

The Texas Gulf Coast    26351
Vail / Beaver Creek     25732
Delaware                20373
Name: region_name_short, dtype: int64

#### Question 8:

#### Total # of check-ins by month in each market

Need assistance completing this in Pandas.

In [26]:
 # Empty

#### Question 9:

#### Calculate the average daily rate (ADR) for all reservations

Mean/Average = $1642 

In [28]:
adr_avg= original['total_paid'].describe()
adr_avg

count    72456.000000
mean      1641.652851
std       1938.210321
min     -42400.000000
25%        563.000000
50%       1071.000000
75%       2099.000000
max      78084.000000
Name: total_paid, dtype: float64

#### Question 10:

#### ADR per region
(Looking at Mean column)

Texas- $928

Delaware- $2351

Vail- $1810

In [29]:
original.groupby(by="region_name_short")["total_paid"].describe()

count         mean          std      min     25%  \
region_name_short                                                          
Delaware              20373.0  2351.074952  1743.770274    -80.0  1219.0   
The Texas Gulf Coast  26351.0   928.372282   739.429614     22.0   467.0   
Vail / Beaver Creek   25732.0  1810.415475  2581.259608 -42400.0   460.0   

                         50%     75%      max  
region_name_short                              
Delaware              2001.0  3002.0  36920.0  
The Texas Gulf Coast   722.0  1167.0  30164.0  
Vail / Beaver Creek    994.0  2220.0  78084.0

#### Question 11: 

#### Was there any notable data cleansing work that you did in order to complete the above question? 

In the uncleaned_vacasa notebook file, I removed any "total_paid" = $0, removed the dollar sign symbols, and removed any NaN's. 